azuredevops/integration_suite_test.go

```diff
@@ -0,0 +1,183 @@
+/*
+Copyright 2023 The Flux authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package azuredevops
+
+import (
+	"context"
+	"fmt"
+	"math/rand"
+	"os"
+	"strings"
+	"testing"
+	"time"
+
+	"encoding/json"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+
+	"github.com/go-logr/logr"
+	"github.com/go-logr/zapr"
+	"go.uber.org/zap"
+
+	"github.com/fluxcd/go-git-providers/gitprovider"
+)
+
+const (
+	azureTokenFile     = "/tmp/azure.token"
+	defaultDescription = "Foo description"
+)
+
+var (
+	azureDomain   = "dev.azure.com"
+	defaultBranch = "main"
+	testOrgName   = "flux"
+	testTeamName  = "fluxcd-test-team"
+	// placeholders, will be randomized and created.
+	testOrgRepoName string
+	testRepoName    string
+	client          gitprovider.Client
+)
+
+func init() {
+	// Call testing.Init() prior to tests.NewParams(), as otherwise -test.* will not be recognised. See also: https://golang.org/doc/go1.13#testing
+	testing.Init()
+	rand.Seed(time.Now().UnixNano())
```

Comment:
Is it really necessary to do this in an `init` func as opposed to `TestProvider` or in the `azure Provider` spec? I personally avoid `init` wherever possible as it's really hard to control or discover when navigating the code.